lib/api/core/models/repositories/roleRepository.js

```diff
@@ -118,6 +118,53 @@ module.exports = function (kuzzle) {
   };
 
   /**
+   * Get from database the document that represent the role given in parameter
+   * @param {Role} role
+   * @returns {Promise}
+   */
+  RoleRepository.prototype.loadRole = function (role) {
+    var deferred = q.defer();
+
+    if (!role._id) {
+      deferred.reject(new BadRequestError('Missing role id'));
+      return deferred.promise;
+    }
+
+    if (this.roles[role._id]) {
+      deferred.resolve(this.roles[role._id]);
+    }
+    else {
+      return this.loadOneFromDatabase(role._id);
```

Comment:
hmm actually, the cacheEngine is NOT undefined (cf line 18 :  cacheEngine: kuzzle.services.list.userCache )

What I don't understand is: why should the cache mechanism be different btw. UserRepository vs. RoleRepository ?
No matter for me anyway, but I suggest that you talk about that with @benoitvidis  to be sure to have an homogeneous code code within each repository prototypes